### PR TITLE
Fix possible BC break in BooleanType

### PR DESCRIPTION
**Pull request by pulzarraider**
Related to #191.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT

### CODE REVIEW

File: Form/Type/BooleanType.php [hunk 1]
```diff
@@ -50,12 +56,26 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
...
+            'translation_domain' => function (Options $options) {
+                if ($options['catalogue']) {
+                    return $options['catalogue'];
+                }
+
+                return $options['translation_domain'];
+            },
+
+            // uncomment in SonataCoreBundle 3.0
+            // 'translation_domain' => 'SonataCoreBundle',
```
Comment (Soullivaneuh) [inline Form/Type/BooleanType.php]: Commented code is note a really good thing IMHO.

Comment (pulzarraider) [inline Form/Type/BooleanType.php]: Do you prefer adding this note in the UPGRADE-3.0.md file created in #192?

Comment (Soullivaneuh) [inline Form/Type/BooleanType.php]: Just a comment like 'Use directly translation_domain in 3.0' would be sufficient.

Comment (pulzarraider) [inline Form/Type/BooleanType.php]: Fixed.

**Comment by Soullivaneuh** [comment]
@pulzarraider Do we have tests for both cases?

**Comment by pulzarraider** [comment]
@Soullivaneuh tests added.

**Pull request merged**